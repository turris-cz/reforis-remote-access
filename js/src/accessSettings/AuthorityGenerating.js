/*
 * Copyright (C) 2019-2021 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, { useEffect } from "react";
import PropTypes from "prop-types";

import { useWSForisModule, Spinner } from "foris";

AuthorityGenerating.propTypes = {
    ws: PropTypes.object.isRequired,
    onAuthoritySuccess: PropTypes.func.isRequired,
};

export const SUCCESS_STATUS = "succeeded";

export default function AuthorityGenerating({ ws, onAuthoritySuccess }) {
    const [generateCA] = useWSForisModule(ws, "remote", "generate_ca");
    useEffect(() => {
        if (generateCA && generateCA.status === SUCCESS_STATUS) {
            onAuthoritySuccess();
        }
    }, [generateCA, onAuthoritySuccess]);

    return (
        <>
            <h2>{_("Generating Certificate Authority")}</h2>
            <Spinner />
            <p>
                {_(`Your certificate authority is now being generated. It \
usually takes a few minutes. Settings will appear here automatically once the \
authority is ready.`)}
            </p>
        </>
    );
}
