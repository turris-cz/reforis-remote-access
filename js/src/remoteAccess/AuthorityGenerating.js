/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, { useEffect } from "react";
import PropTypes from "prop-types";

import { useWSForisModule, Spinner } from "foris";

AuthorityGenerating.propTypes = {
    ws: PropTypes.object.isRequired,
    onSuccess: PropTypes.func.isRequired,
};

export const SUCCESS_STATUS = "succeeded";

export default function AuthorityGenerating({ ws, onSuccess }) {
    const [generateCA] = useWSForisModule(ws, "remote", "generate_ca");
    useEffect(() => {
        if (generateCA && generateCA.status === SUCCESS_STATUS) {
            onSuccess();
        }
    }, [generateCA, onSuccess]);

    return (
        <>
            <h3>{_("Generating certification authority")}</h3>
            <Spinner />
            <p>{_("Your certification authority is now being generated. It usually takes a few minutes. Settings will appear here automatically once the authority is ready.")}</p>
        </>
    );
}
