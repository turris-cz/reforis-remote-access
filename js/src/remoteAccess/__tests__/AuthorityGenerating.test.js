/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { render, act } from "foris/testUtils/customTestRender";

import { WebSockets } from "foris";

import AuthorityGenerating, { SUCCESS_STATUS } from "../AuthorityGenerating";

describe("<AuthorityGenerating />", () => {
    const onSuccess = jest.fn();
    let webSockets,
        componentContainer;

    beforeEach(() => {
        webSockets = new WebSockets();
        const { container } = render(<AuthorityGenerating ws={webSockets} onSuccess={onSuccess} />);
        componentContainer = container;
    });

    it("should match snapshot", () => {
        expect(componentContainer).toMatchSnapshot();
    });

    it("should call onSuccess when CA is ready", () => {
        expect(onSuccess).not.toBeCalled();
        const wsMessage = { module: "remote", action: "generate_ca", data: {status: SUCCESS_STATUS} };
        act(() => webSockets.dispatch(wsMessage));
        expect(onSuccess).toHaveBeenCalledTimes(1);
    });
});
