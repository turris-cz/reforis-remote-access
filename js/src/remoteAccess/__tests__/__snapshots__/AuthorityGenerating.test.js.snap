// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorityGenerating /> should match snapshot 1`] = `
<div>
  <h3>
    Generating certificate authority
  </h3>
  <div
    class="spinner-wrapper my-3 text-center"
  >
    <div
      class="spinner-border "
      role="status"
    >
      <span
        class="sr-only"
      />
    </div>
    <div
      class="spinner-text"
    />
  </div>
  <p>
    Your certificate authority is now being generated. It usually takes a few minutes. Settings will appear here automatically once the authority is ready.
  </p>
</div>
`;
